exporter.go

```diff
@@ -1137,6 +1237,109 @@ func getKeyInfo(c redis.Conn, key string) (info keyInfo, err error) {
 	return info, err
 }
 
+func getStreamInfo(c redis.Conn, key string) (streamInfo, error) {
+	var info streamInfo
+	keyType, err := redis.String(doRedisCmd(c, "TYPE", key))
+	if err != nil {
+		return info, err
+	}
+
+	switch keyType {
+	case "none":
+		return info, errNotFound
+	case "stream":
+		info, err = scanStream(c, key)
+		if err != nil {
+			return info, err
+		}
+	default:
+		return info, errNotStream
+	}
+	return info, err
+}
+
+func scanStream(c redis.Conn, key string) (streamInfo, error) {
+	var stream streamInfo
+	v, err := redis.Values(doRedisCmd(c, "XINFO", "STREAM", key))
+	if err != nil {
+		return stream, err
+	}
+	// Scan slice to struct
+	err = redis.ScanStruct(v, &stream)
+	if err != nil {
+		return stream, err
+	}
+
+	stream.StreamGroupsInfo, err = scanStreamGroups(c, key)
+	if err != nil {
+		return stream, err
+	}
+
+	log.Debugf("stream: %#v", &stream)
+	return stream, nil
+}
+
+func scanStreamGroups(c redis.Conn, stream string) ([]streamGroupsInfo, error) {
+
+	groups, err := redis.Values(doRedisCmd(c, "XINFO", "GROUPS", stream))
+	if err != nil {
+		return nil, err
+	}
+
+	var result []streamGroupsInfo
+	for _, g := range groups {
+
+		v, err := redis.Values(g, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		var group streamGroupsInfo
+		err = redis.ScanStruct(v, &group)
+		if err != nil {
+			return nil, err
+		}
+
+		group.StreamGroupConsumersInfo, err = scanStreamGroupConsumers(c, stream, group.Name)
+		if err != nil {
+			return nil, err
+		}
+
+		result = append(result, group)
+	}
+
+	log.Debugf("groups: %v", result)
+	return result, nil
+}
+
+func scanStreamGroupConsumers(c redis.Conn, stream string, group string) ([]streamGroupConsumersInfo, error) {
+	consumers, err := redis.Values(doRedisCmd(c, "XINFO", "CONSUMERS", stream, group))
+	if err != nil {
+		return nil, err
+	}
+
+	var result []streamGroupConsumersInfo
+	for _, c := range consumers {
+
+		v, err := redis.Values(c, nil)
+		if err != nil {
+			return nil, err
+		}
+		log.Debugf("v: %#v", v)
+
+		var consumer streamGroupConsumersInfo
+		err = redis.ScanStruct(v, &consumer)
+		if err != nil {
+			return nil, err
```

Comment:
Does it make sense to completely abort here or just log the error and continue to the next item in `consumers`?

Reply:
Agree, `continue` is more preferred.